### PR TITLE
Tools: Testbench: Add capture direction and multi-pipeline run

### PR DESCRIPTION
**Pull request by singalsu**

### CODE REVIEW

File: tools/testbench/common_test.c [hunk 1]
```diff
@@ -144,24 +138,16 @@ int tb_pipeline_start(struct ipc *ipc, struct pipeline *p,
...
-		return -EINVAL;
-	}
-
-	/* Point to pipeline component device */
-	cd = pcm_dev->cd;
+	/* Get pipeline host component */
+	cd = tb_get_pipeline_host(p);
 
-	ret = pipeline_trigger(p, cd, COMP_TRIGGER_STOP);
+	// fprintf(stderr, "Stopping pipeline_id %d\n", p->pipeline_id);
```
Comment (singalsu) [inline tools/testbench/common_test.c:149]: Remove // line

File: tools/testbench/testbench.c [hunk 2]
```diff
@@ -659,14 +764,21 @@ static void *pipline_test(void *data)
 		printf("		           Test Start %d\n", dp_count);
 		printf("==========================================================\n");
 
-		err = test_pipeline_load(ptdata, &ctx, tp->pipelines[0]);
+		err = test_pipeline_load(ptdata, &ctx, tp->pipelines[0]); //fixme, last arg
```
Comment (singalsu) [inline tools/testbench/testbench.c:767]: Fix // and unused argument

File: tools/testbench/topology.c [hunk 3]
```diff
@@ -779,14 +784,24 @@ int parse_topology(struct tplg_context *ctx)
...
-		if (hdr->index != ctx->pipeline_id) {
+		pipeline_match = false;
+		for (i = 0; i < tp->pipeline_num; i++) {
+			if (hdr->index == tp->pipelines[i]) {
+				pipeline_match = true;
+				break;
+			}
+		}
+
+		//pipeline_match = true; // Hack to get AEC reference connect included
```
Comment (singalsu) [inline tools/testbench/topology.c:795]: Wonder if need for mode where everything in topology is tried to be instantiated, not just -p x,y,z. Remove //.

File: tools/testbench/common_test.c (hunk 1 above)
Comment (lgirdwood) [reply tools/testbench/common_test.c:149]: ping

**Review by lgirdwood — COMMENTED** [review]
This is an ABI update, we should align the kernel ABI file too (eben though it's not used by kernel).

**Comment by sys-pt1s** [comment]
Can one of the admins verify this patch?

**Comment by lgirdwood** [comment]
@singalsu do you have a PR that syncs the kernel version of this header ?

**Comment by kv2019i** [comment]
@lgirdwood wrote:
> @singalsu do you have a PR that syncs the kernel version of this header ?

Just acking here as well, that as per current process, this is not needed to proceed with this merge. As discussed, we might go back to old procedure.

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
SOFCI TEST

**Pull request merged**